### PR TITLE
Add more docs

### PR DESCRIPTION
**Pull request by jveski**

### CODE REVIEW

File: dev/build.sh [hunk 1]
```diff
@@ -11,8 +11,8 @@ export TAG="$(date +%s)"
 
 function build() {
     cmd=$(basename $1)
-    buildah build -t "$REGISTRY/$cmd:$TAG" -f "$f/Dockerfile"
-    buildah push "$REGISTRY/$cmd:$TAG"
+    docker build --quiet -t "$REGISTRY/$cmd:$TAG" -f "$f/Dockerfile" .
```
Comment (jveski) [inline dev/build.sh:14]: Now the script is compatible with both buildah and docker

File: api/v1/config/crd/eno.azure.io_synthesizers.yaml [hunk 2]
```diff
@@ -21,39 +21,12 @@ spec:
     name: v1
     schema:
       openAPIV3Schema:
-        description: "Synthesizers are any process that can run in a Kubernetes container
```
Comment (jveski) [inline api/v1/config/crd/eno.azure.io_synthesizers.yaml:24]: Some of the more details godoc comments have been moved to `reference.md`

File: docs/reference.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+This example will cause a "synthesis" of the composition through its synthesizer
+and track the lifecycle of the resulting resources:
+deleting the composition will delete the resources,
+updating it will cause resynthesis, etc.
+
+### Deletion Modes
+
+To keep any resources created because of a composition,
+Eno supports an orphaning deletion mode by setting this annotation:
```
Comment (AYM1607) [inline docs/reference.md:25]: Do you think the phrasing here is clear enough to indicate that the annotation should go on the composition?

File: docs/reference.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+```yaml
+eno.azure.io/deletion-strategy: orphan
+```
+
+## Synthesizer Basics
+
+Synthesizers are container images that have one or more implementations of the [KCM Functions API](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/docs/api-conventions/functions-spec.md).
```
Comment (AYM1607) [inline docs/reference.md:33]: KCM -> KRM

File: docs/reference.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
+# Reference Docs
```
Comment (AYM1607) [inline docs/reference.md:1]: Nit: Missing symphonies, but they can go on a separate PR 😄
